features/workSession/components/layouts/TabArea/TabArea.tsx

```diff
@@ -149,30 +145,22 @@ const TabArea = ({
     triggerPosition: editTabMenuPopoverTriggerPosition,
   } = usePopover();
 
-  // on selecting a tab
-  const handleSelectTab = (tab: Tab) => {
-    dispatch(updateSelectedTab(tab));
-  };
-
-  useEffect(() => {
-    if (tabs.length) dispatch(updateSelectedTab(tabs[0]));
-  }, [dispatch, tabs]);
-
-  // Close both popover and modal when selected tab is changed
-  useEffect(() => {
-    onCloseEditTabMenuPopover();
-    onCloseRenameTabModal();
-  }, [selectedTab]);
-
   // Close popover when modal is opened
   useEffect(() => {
     if (isOpenRenameTabModal) {
       onCloseEditTabMenuPopover();
     }
   }, [isOpenRenameTabModal, onCloseEditTabMenuPopover]);
 
+  // global state `selectedTab` is not a state to manage tab data,
+  // so we need to find the tab data from useState `tabs` array
+  // There's a slight update timing gap on useState and global state,
+  // so we need to set tabs[0] as default value
+  const selectedTabState =
+    tabs.find((tab) => tab.id === selectedTab.id) ?? tabs[0];
+
```

Comment:
Nice workaround!!
We need to rethink how to manage the Tab-List-Task states but for now this is a really good solution.